libraries/AP_AHRS/AP_AHRS_DCM.cpp

```diff
@@ -81,6 +84,13 @@ AP_AHRS_DCM::update(bool skip_ins_update)
     // Perform drift correction
     drift_correction(delta_t);
 
+    if(_ins.get_gyro().z>GYRO_MAX_SPIN_RATE*M_PI/180.0 || _ins.get_gyro().z<-GYRO_MAX_SPIN_RATE*M_PI/180.0)
```

Comment:
please use the built in function radians(GYRO_MAX_SPIN_RATE) and do it once. Also, might want to make it a const and performed once since it never changes
https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Math/AP_Math.h#L157

In fact, for simplicty, can you change it to:
if (absf(_ins.get_gyro().z) > GYRO_MAX_SPIN_RATE_RAD)

note, the units are in the name so we don't have to guess and the define does the translation for us once at compile time.

Reply:
ok for the buildin function. But when I try rebase, I have the following error
git rebase -i HEAD~20
error: object file .git/objects/3d/7468628975948f8da99788f2b45805475e5600 is empty
fatal: loose object 3d7468628975948f8da99788f2b45805475e5600 (stored in .git/objects/3d/7468628975948f8da99788f2b45805475e5600) is corrupt
[HEAD détachée 
Could not apply c70e2a1999d2bffc53ca1edee551d1b1c46aa54c... AP_Math : Quaternion slight addon

sorry but I'm not a git expert

Reply:
that command rebases back 20 commits. What you want to do is rebase onto master

Reply:
plane could not be considered as in our case, our drone is only a rotary wing, it would not make sense to use plane for it.

Reply:
make sure to fetch the latest upstream master first :)